analyzer/src/main/kotlin/managers/MavenDependencyHandler.kt

```diff
@@ -0,0 +1,123 @@
+
+import org.eclipse.aether.graph.DependencyNode
+
+import org.ossreviewtoolkit.analyzer.managers.utils.DependencyHandler
+import org.ossreviewtoolkit.analyzer.managers.utils.MavenSupport
+import org.ossreviewtoolkit.analyzer.managers.utils.identifier
+import org.ossreviewtoolkit.model.OrtIssue
+import org.ossreviewtoolkit.model.Package
+import org.ossreviewtoolkit.model.PackageLinkage
+import org.ossreviewtoolkit.model.createAndLogIssue
+import org.ossreviewtoolkit.utils.collectMessagesAsString
+import org.ossreviewtoolkit.utils.log
+import org.ossreviewtoolkit.utils.showStackTrace
+
+/**
+ * A specialized [DependencyHandler] implementation for the dependency model of Maven.
+ */
+class MavenDependencyHandler(
+    /** The name of the associated package manager. */
+    val managerName: String,
+
+    /** The helper object to invoke Maven-related functionality. */
+    val support: MavenSupport,
+
+    /**
+     * A map with information about the local projects in the current Maven build. Dependencies pointing to projects
+     * sometimes need to be treated in a special way.
+     */
+    val localProjects: Map<String, MavenProject>,
+
+    /**
+     * A flag whether [SBT compatibility mode][Maven.enableSbtMode] is enabled.
+     */
+    val sbtMode: Boolean
+) : DependencyHandler<DependencyNode> {
+    override fun identifierFor(dependency: DependencyNode): String {
+        val id = dependency.identifier()
+        val typePrefix = if (isLocalProject(id)) managerName else "Maven"
+        return "$typePrefix:$id"
+    }
+
+    override fun dependenciesFor(dependency: DependencyNode): Collection<DependencyNode> {
+        val filteredChildren = dependency.children.filterNot { node ->
+            TOOL_DEPENDENCIES.any(node.artifact.identifier()::startsWith)
+        }
+
+        if (filteredChildren.size < dependency.children.size) {
+            log.info { "Omitting the Java < 1.9 system dependency on 'tools.jar'." }
+        }
+
+        return filteredChildren
+    }
+
+    override fun linkageFor(dependency: DependencyNode): PackageLinkage =
+        if (isLocalProject(dependency)) PackageLinkage.PROJECT_DYNAMIC else PackageLinkage.DYNAMIC
+
+    override fun createPackage(
+        identifier: String,
+        dependency: DependencyNode,
+        issues: MutableList<OrtIssue>
+    ): Package? =
+        if (isLocalProject(dependency)) {
+            null
+        } else {
+            try {
+                support.parsePackage(dependency.artifact, dependency.repositories, localProjects, sbtMode)
+            } catch (e: ProjectBuildingException) {
+                e.showStackTrace()
+
+                issues += createAndLogIssue(
+                    source = managerName,
+                    message = "Could not get package information for dependency '" +
+                            "${dependency.artifact.identifier()}': ${e.collectMessagesAsString()}"
+                )
+
+                null
+            }
```

Comment:
I haven't tried whether it would look really nicer, but how about a `runCatching` / `onFailure` / `getOrNull` chain instead?

Reply:
Yes, this looks more stylish. Changed.